src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs

```diff
@@ -304,10 +304,12 @@ protected virtual bool IsPayloadAttributesTimestampValid(Block newHeadBlock, For
     private ResultWrapper<ForkchoiceUpdatedV1Result> StartBuildingPayload(Block newHeadBlock, ForkchoiceStateV1 forkchoiceState, PayloadAttributes? payloadAttributes)
     {
         string? payloadId = null;
+        bool isPayloadSimulated = false;
 
-        if (_simulateBlockProduction)
+        if (_simulateBlockProduction && payloadAttributes is null)
         {
-            payloadAttributes ??= newHeadBlock.Header.GenerateSimulatedPayload();
+            isPayloadSimulated = true;
```

Comment:
IMO simpler to add after line 323:

`payloadId = _simulateBlockProduction ? null : payloadId;`